test/report/test-report-exclude-network.js

```diff
@@ -38,4 +39,60 @@ describe('report exclude network option', () => {
     const report = process.report.getReport();
     assert.strictEqual(report.header.networkInterfaces, undefined);
   });
+
+  it('should not do DNS queries in libuv if exclude network', async () => {
+    const server = http.createServer(function(req, res) {
+      res.writeHead(200, { 'Content-Type': 'text/plain' });
+      res.end();
+    });
+    let ipv6Available = true;
+    const port = await new Promise((resolve) => server.listen(0, async () => {
+      await Promise.all([
+        fetch('http://127.0.0.1:' + server.address().port),
+        fetch('http://[::1]:' + server.address().port).catch(() => ipv6Available = false),
+      ]);
+      resolve(server.address().port);
+      server.close();
+    }));
+    process.report.excludeNetwork = false;
+    let report = process.report.getReport();
+    let tcp = report.libuv.filter((uv) => uv.type === 'tcp' && uv.remoteEndpoint?.port === port);
+    assert.strictEqual(tcp.length, ipv6Available ? 2 : 1);
+    const findHandle = (local, ip4 = true) => {
+      return tcp.find(
+        ({ [local ? 'localEndpoint' : 'remoteEndpoint']: ep }) =>
+          (ep[ip4 ? 'ip4' : 'ip6'] === (ip4 ? '127.0.0.1' : '::1')),
+      )?.[local ? 'localEndpoint' : 'remoteEndpoint'];
+    };
+    try {
+      // The reverse DNS of 127.0.0.1 can be a lot of things other than localhost
+      // it could resolve to the server name for instance
+      assert.notStrictEqual(findHandle(true)?.host, '127.0.0.1');
+      assert.notStrictEqual(findHandle(false)?.host, '127.0.0.1');
+
+      if (ipv6Available) {
+        assert.notStrictEqual(findHandle(true, false)?.host, '::1');
+        assert.notStrictEqual(findHandle(false, false)?.host, '::1');
+      }
+    } catch (e) {
+      throw new Error(e.message + ' in ' + JSON.stringify(tcp, null, 2));
```

Comment:
```suggestion
      throw new Error(e?.message + ' in ' + JSON.stringify(tcp, null, 2), { cause: e });
```

Do we need to try/catch here, or could we use the third argument of the `assert.notStrictEqual` methods?

Reply:
The third argument would just mean a 4 times duplication which is why I didn't use it